thirtyinch/src/main/java/net/grandcentrix/thirtyinch/internal/PresenterScope.java

```diff
@@ -54,13 +58,19 @@ public boolean isEmpty() {
         return mStore.isEmpty();
     }
 
-    public TiPresenter remove(final String id) {
+    public TiPresenter remove(@NonNull final String id) {
         final TiPresenter presenter = mStore.remove(id);
         TiLog.d(TAG, "remove " + id + " " + presenter);
         return presenter;
     }
 
-    public void save(final String id, final TiPresenter presenter) {
+    public void save(@NonNull final String id, @NonNull final TiPresenter presenter) {
+        if (id == null) {
```

Comment:
I don't know if I agree with that 🤔 
According to the `@NonNull` doc:
```
 Denotes that a parameter, field or method return value can never be null.
```

Which means. If we put `null` in here, we do something wrong.
Of course, such issues  may happen with mutation testing ( 😉  ) but normally it doesn't by a human (if we add the annotation)....

... 🤔

Reply:
Then we should also remove other checks like those in [`TiPresenter`](https://github.com/grandcentrix/ThirtyInch/blob/master/thirtyinch/src/main/java/net/grandcentrix/thirtyinch/TiPresenter.java#L158)

For me the `@NonNull` annotation is just a warning: "Don't put null here, it won't work".
The problem here is that it will work in a way which is not intended. That's why I want to explicitly disallow this option by throwing the Exception. The annotation is just an additional hint which helps devs to get a warning before running into the runtime exceptions.

Btw, Kotlin does the same. When a parameter is non nullable the compile adds a non null check at the beginning of the method in byte code.